File: back/back/domain/entities/ITela.cs
using System;

namespace back.domain.entities
{
    public interface ITela
    {
        public string Nome { get; set; }
        public string Url { get; set; }
        public string AddUrl { get; set; }
        public string Target { get; set; }
        public bool Nivel { get; set; }
        public Int16 Ordem { get; set; }
        public string Modulo { get; set; }
        public bool Sd { get; set; }
        public string ImagemSd { get; set; }
        public string IconClass { get; set; }
        public Nullable<int> TelaId { get; set; }
    }
}

File: front/src/app/Domain/Models/ITela.ts
/**
 * @class Tela
 * @implements {ITela}
 * @classdesc Classe que representa uma tela.
 * @constructor
 * @param {string} nome Nome da tela.
 * @param {string} descricao Descrição da tela.
 * @param {string} url Url da tela.
 * jao é um jogador de lol
 */
export interface ITela {
  id: number | undefined;
  nome: string;
  url: string;
  addUrl: string;
  target: string;
  nivel: boolean;
  ordem: number;
  modulo: string;
  sd: boolean;
  imagemSd: string;
  iconClass: string;
  telaId: number | undefined;
  relateds: ITela[];
  tela: ITela | null;
}
